.github/workflows/nextflow-tests.yml

```diff
@@ -73,11 +169,257 @@ jobs:
 
   summary:
     runs-on: ubuntu-latest
-    needs: run
-    if: ${{ !cancelled() }}
+    needs: [run, discover]
+    if: ${{ !cancelled() && needs.discover.result == 'success' && needs.discover.outputs.abort_early == 'false' }}
 
     steps:
       - uses: actions/github-script@v7
-        if: ${{ needs.run.result != 'success' && needs.run.result != 'skipped' }}
+        id: status
+        name: Determine required steps
         with:
-          script: core.setFailed('Tests failed!')
+          script: |
+            var needs = ${{ toJSON(needs) }}
+            var tests_failed = needs.run.result != 'success' &&
+              needs.run.result != 'skipped'
+
+            var should_comment = tests_failed &&
+                !(needs.discover.outputs.apply_fixes == 'true')
```

Comment:
Why `!(... == 'true')` rather than `(... != 'true')

Reply:
I was so deep underwater at the time: https://github.com/uclahs-cds/tool-Nextflow-action/commit/0b118038cb31f072646584cf80636eca73a79dfd

I'll clean it up.

Reply:
Been there. :)

Reply:
Done with e7c93c7.